### DISCUSSION TITLE
[Feature Request] Expansion of Things and Home Assistant

### DISCUSSION THREAD

**Discussion by ados8** [donetick/donetick#64] [Ideas]
Things is the basis for a really powerful functionality set.
I assume this is being expanded on but noticed Home Assistant (HA) integration is not.
Wanted to bring up some ideas/suggestions and how HA is likely the best option to manage Things.

Things could be expanded in both functionality within Donetick and through HA integration.
Currently the only method in which Thing states change is manually, even the completion of tasks doesn't allow state change.
The biggest adoption issues that task apps have is time friction, that is the time investment in managing the tasks themselves.
If a user needs to train their family to manually update the states of variables they've lost.

In further testing of Donetick I noticed you cannot have a recurring task and with Thing state attached.
For example, a shopping list is repetitive and with a Boolean it works to trigger the due date but once complete it's gone.
It subsequently doesn't allow setting the Boolean to another value so a user must go into Things and set back to false and then make another task.

There's 2 ways I can see to resolve this, build management functionality into Donetick to automate states or offload it to HA.
Requiring people to have HA would be tough but at least lowers the development burden to build out 2 systems or even Donetick.
HA would instantly provide a massive capability set, suddenly automations, scripts, voice assistants and more is possible in Donetick.
HA could control Donetick variables and have far more complicated flows if needed.
Instead of one Thing triggering a task HA scripts can have if this and this but not this and only this then make task etc.
To have this work Donetick would need a full API for HA to not only pull information such as tasks, sensors/variables, users etc. but manage Donetick.

**Comment by ab3lson** [comment]
This was my first thought when I saw the “Things” tab! At the very least, it would be nice to be able to control a Thing value via an API. I can think of a bunch of use cases where something in Home Assistant triggers a Thing value to update and schedules a new task.

Edit: It looks like there already is an API for updating Thing values. I can’t find any API docs, but I should be able to figure out what the request wants from looking at the [code](https://github.com/donetick/donetick/blob/main/internal/thing/handler.go#L78). For the time being, you can use this to update values using Home Assistant automations or NodeRed.
